app/common/elasticsearchclient.py

```diff
@@ -240,19 +240,25 @@ def best_hit_search(self,
         # current_app.logger.debug("Got %d Hits in %ims" % (res['hits']['total'], res['took']))
         data = []
         
-        for res in results['responses']:
-            res_data = [];
-            for hit in res['hits']['hits']:
```

Comment:
Yes, I've seen that you are doing that in the _filter_by_target_list_ feature, but I think that is not correct.
So, when an ensembl id is provided you are not firing a search with it, so what happens if the user upload an invalid ensembl id? (outdated?, two ens ids together?). You should still make a search with them to make sure that the ensid is valid